app/operations/programs/update.rb

```diff
@@ -0,0 +1,27 @@
+module Programs
+  class Update < Operation
+    integer :id
+
+    hash :attributes do
+      string :name
+      boolean :custom, default: false
+      string :description, default: ""
+      hash :metadata, strip: false, default: {}
+    end
+
+    relationships do
+      has_many :classrooms
+    end
+
+    def execute
```

Comment:
you're not wrong, hound, but I'm pretty sure this is how ActiveInteraction is supposed to work. Otherwise, optional attributes will get overwritten by their defaults if they're missing from the params.